src/components/table/table_root.tsx

```diff
@@ -28,30 +31,33 @@ import { useTableColumns } from './use_table_columns.js';
 import { useTableScroll } from './use_table_scroll.js';
 
 const CustomHTMLTable = styled(HTMLTable, {
-  shouldForwardProp: (prop) => prop !== 'striped' && prop !== 'stickyHeader',
-})<{ stickyHeader: boolean }>`
+  shouldForwardProp: (prop) =>
+    prop !== 'striped' && prop !== 'stickyHeader' && prop !== 'noHeader',
```

Comment:
This start to be hard to understand maybe we can refactor this to a function or something else ?